src/main/java/fi/helsinki/opintoni/integration/studyregistry/sisu/SisuGraphQLClient.java

```diff
@@ -17,65 +17,154 @@
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.EventOverrideResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.LocalizedStringResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.LocationResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.OrganisationResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.PrivatePersonResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.Private_personQueryRequest;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.Private_personQueryResponse;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.Private_personsQueryRequest;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.PublicPersonResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.StudyEventRealisationResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.StudyEventResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.StudyGroupSetResponseProjection;
+import fi.helsinki.opintoni.integration.studyregistry.sisu.model.StudySubGroupResponseProjection;
+import fi.helsinki.opintoni.util.DateTimeUtil;
 
 public class SisuGraphQLClient implements SisuClient {
-    private final String endPointURL;
 
-    private static final Logger log = LoggerFactory.getLogger(SisuGraphQLClient.class);
+    private final RestTemplate restTemplate;
+    private final String url;
 
-    public SisuGraphQLClient(String endPointURL) {
-        this.endPointURL = endPointURL;
+    public SisuGraphQLClient(String url, RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+        this.url = url;
     }
 
-    @Override
-    public PrivatePersonRequest getPrivatePerson(String id) {
-
-        return queryWithPrivatePersonId(id, PrivatePersonRequest.class);
+    private <T> T execute(GraphQLRequest request, Class<T> type) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
 
+        HttpEntity<String> httpEntity = new HttpEntity<>(request.toHttpJsonBody(), headers);
+        return restTemplate.exchange(url, HttpMethod.POST, httpEntity, type).getBody();
     }
 
-    @Override
-    public StudyAttainmentRequest getStudyAttainments(String id) {
-        return queryWithPrivatePersonId(id, StudyAttainmentRequest.class);
-    }
+    private static final LocalizedStringResponseProjection ALL_LANGUAGES = new LocalizedStringResponseProjection().en().fi().sv();
 
-    private <T> T queryWithPrivatePersonId(String id, Class<T> type) {
-        return queryWithArguments(new Arguments("private_person", new Argument<>("id", getSisuPrivatePersonId(id))), type);
+    @Override
+    @Cacheable(value = CacheConstants.GRAPHQL_CUR_SEARCH, cacheManager = "transientCacheManager", sync = true)
```

Comment:
Kakkuavainta ei ole speksattu, joten se on kaikki metodiparametrit, ml. LocalDate since, joka on joka kutsulla eri.

Reply:
ei kai since vaihdu joka kutsulla?

Reply:
sisuClient.curSearch(teacherNumber, LocalDate.now(ZoneId.of("Europe/Helsinki")))
ja
sisuClient.curSearch(hloId, LocalDate.now())